src/validation/rules/KnownTypeNames.js

```diff
@@ -21,9 +21,22 @@ export function unknownTypeMessage(type: any): string {
  *
  * A GraphQL document is only valid if referenced types (specifically
  * variable definitions and fragment conditions) are defined by the type schema.
+ *
+ * We ignore the types that are referenced by types that are not part of the
+ * schema.
  */
 export function KnownTypeNames(context: ValidationContext): any {
   return {
+    ObjectTypeDefinition(node) {
```

Comment:
We shouldn't be validating client side type definitions at all, presently. I think you can safely return false without any other logic here. Similar will be necessary for other kinds of type definitions.